src/block.rs

```diff
@@ -1,34 +1,118 @@
 //use std::stream::Stream;
 //use futures::io::AsyncRead;
 
-#[derive(Debug)]
+use serde::{Serialize, Deserialize};
+
+use crate::burnfee::BurnFee;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
 pub struct Block {
-    
+    pub body:     BlockBody,
+    pub is_valid: u8,
+    mintid:   u32,
+    maxtid:   u32,
```

Comment:
I think we use maxtid when creating blocks. The block creator basically needs to know what tid they should use when they start putting things into blocks. it is probably easier to keep it than keep the last TX around and check the ID that way.

Right now mintxid is used in the JS version by the explorer (?). It was added so that the core blockchain.js code could find out what block a particular transaction would be in. And while it might be useful for the engine to be able to scroll through blocks and know where a tx is located, but i don't think we need it for consensus and we can probably drop it in rust if we want.